components/List/List.js

```diff
@@ -1,10 +1,26 @@
 import { StyledList, StyledItem } from "@/styles";
 
-export default function List({ transactions }) {
+function filteredTransactions(transactions, transactionFilter) {
+  if (transactionFilter === "all") {
+    return transactions;
+  } else {
+    return transactions.filter(
+      (transaction) => transaction.type === transactionFilter
+    );
+  }
+}
```

Comment:
Please check the logic again.  💡

The concept is as follows: 
- Initial state is set to "all".
- Clicking on "income" or "expense" updates the state.
- Transactions are filtered by the selected type.